classes/plugins/Plugin.inc.php

```diff
@@ -391,9 +391,17 @@ public function _overridePluginTemplates($hookName, $args) {
 
 		// Check if an overriding plugin exists in the plugin path.
 		$checkPluginPath = sprintf('%s/%s', $this->getPluginPath(), $checkFilePath);
+
+		// Parent template path if exists
+		if (isset($this->parent)) {
+			$checkParentPath = $this->parent->getTemplatePath() . DIRECTORY_SEPARATOR . $template;
+		}
+
 		if (file_exists($checkPluginPath)) {
 			$filePath = $checkPluginPath;
-		// Backward compatibility for OJS prior to 3.1.2; changed path to templates for plugins.
+		} elseif ($checkParentPath && file_exists($checkParentPath)) {
+			$filePath = $checkParentPath;
+			// Backward compatibility for OJS prior to 3.1.2; changed path to templates for plugins.
```

Comment:
Cool, I _think_ this is the right place to do this, but let's do a recursive check of all parents, and lets limit the `parent` check to `ThemePlugin` objects. Here's a recursive function we can add to Plugin.inc.php:

```php
/**
 * @return string|false
 */
private function _findOverriddenTemplate($path) {
  $fullPath = sprintf('%s/%s', $this->getPluginPath(), $path);

  if (file_exists($fullPath)) {
    return $fullPath;
  }

  if (is_a($this, 'ThemePlugin')
      && $this->parent
      && $fullPath = $this->parent->_findOverriddenTemplate($path)) {
    return $fullPath;
  }

  return null;
}
```

And then the conditional logic here (line 393-405) can be simplified to:

```
$filePath = $this->_findOverriddenTemplate($checkFilePath);
if (!$filePath) {
  // back compat fallback
  $filePath = ...;
}
```

Reply:
Hmm, looks like $filePath to this method is passed by reference, so it should be returned as it was if checks are failed. And probably compatibility check should be taken into account because it gives me false positive results with this recursion.

Reply:
:+1: yeah that was just some sample code I haven't actually tested it.